### PR TITLE
#478 - Create tests for Datasource.Close()

### PR DESCRIPTION
**Pull request by fredyk**
Issue: https://github.com/fredyk/westack-go/issues/478

### CODE REVIEW

File: westack/tests/westack_datasource_test.go [hunk 1]
```diff
@@ -43,6 +43,19 @@ func Test_Datasource_Initialize_ConnectError(t *testing.T) {
...
+func Test_DatasourceClose(t *testing.T) {
+
+	t.Parallel()
+
+	ds, err := app.FindDatasource("db_expected_to_be_closed")
+	assert.NoError(t, err)
+	assert.NotNil(t, ds)
+
+	err = ds.Close()
+	assert.NoError(t, err)
```
Comment (tanryberdi) [inline westack/tests/westack_datasource_test.go:55]: I think to check `err` for nil is not enough for this situation, you must also check for `FindDataSource`, the result of this also must be `nil`, wdyt?

Comment (tanryberdi) [reply westack/tests/westack_datasource_test.go:55]: @fredyk, it is good to check `app.FindDatasource()` as well. Let me add one test example to explain thorough :
`func (s *TestSuite) TestDeleteUser() {
	var dateNow = time.Now().UTC()

	ctx, storageWriter, storageReader := s.ctx, s.storageWriter, s.storageReader

	input := users.NedapUser{
		PK:        "mock_pk",
		Email:     "mock_email",
		Hash:      "mock_hash",
		UpdatedAt: dateNow,
	}

	err := storageWriter.PutUser(ctx, input)
	s.Nil(err)

	// We are deleting user with id "mock_id"
	err = storageWriter.DeleteUser(ctx, "mock_pk")
	s.Nil(err)

	// We have no any user after deleting
	m, err := storageReader.GetAll(ctx)
	s.Nil(err)
	s.Assert().Empty(m)
}`

in here we deleting user with `DeleteUser()` function and return `error`, for above test case it is not enough to check only `error` is nil or not, plus, we need to check deleted user is exist or not in. So, I suggested to test function itself.

File: westack/tests/westack_datasource_test.go [hunk 2]
```diff
@@ -54,6 +54,13 @@ func Test_DatasourceClose(t *testing.T) {
 	err = ds.Close()
 	assert.NoError(t, err)
 
+	// Based on suggestion from @tanryberdi: https://github.com/fredyk/westack-go/pull/480#discussion_r1312634782
+	// Attempt to perform a query. We don't mind the queried collection because the client
+	// is disconnected anyway
+	result, err := ds.FindMany("unknownCollection", nil)
+	assert.Errorf(t, err, "client is disconnected")
+	assert.Nil(t, result)
```
Comment (fredyk) [inline westack/tests/westack_datasource_test.go:62]: @tanryberdi Do you think ds.Close() is well tested now with these additional checks?

Comment (tanryberdi) [reply westack/tests/westack_datasource_test.go:62]: yes @fredyk, it was great now, sorry for disturbing :D, btw, I will try to add test case for `FindDataSource` functions, you can assign to me

Comment (fredyk) [reply westack/tests/westack_datasource_test.go:62]: Great! Just assigned that other one #493 to you. Thanks very much for your collaboration :smiley:

**Review by gmontielg — APPROVED** [review]

**Pull request merged**